src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8909,13 +8909,22 @@ export class StudyViewPageStore
             }
         }
 
-        let url = '/';
-        if (!_.isEmpty(this.geneQueries)) {
-            formOps.Action = 'Submit';
-            formOps.gene_list = this.geneQueries
-                .map(query => unparseOQLQueryLine(query))
-                .join('\n');
-            url = '/results';
+        if (url) {
+            if (!_.isEmpty(this.geneQueries)) {
+                formOps.Action = 'Submit';
```

Comment:
this code is duplicated below and its presence muddies the conditional.    i would handle the case where there is no url using a default value for the url arg 
```
onSubmitQuery(url:string="/"){...}
```
then you can just check to see if there's geneQueries and update url to results